### PR TITLE
Update and organize Configurations.md

### PR DESCRIPTION
**Pull request by TheTechn0mancer**
Fixed lexicographic ordering errors, removed deprecated options, removed one self-referential "See also", fixed formatting on sub-header.

### CODE REVIEW

**Comment by TheTechn0mancer** [comment]
Let me fix the issue with group_imports

File: Configurations.md [hunk 1]
```diff
@@ -529,6 +529,14 @@ Note that this option may be soft-deprecated in the future once the [ignore](#ig
...
 - **Stable**: Yes
 
+## `doc_comment_code_block_width`
+
+Max width for code snippets included in doc comments. Only used if [`format_code_in_doc_comments`](#format_code_in_doc_comments) is true.
+
+- **Default value**: `100`
+- **Possible values**: any positive integer that is less than or equal to the value specified for [`max_width`](#max_width)
+- **Stable**: No (tracking issue: [#5359](https://github.com/rust-lang/rustfmt/issues/5359))
+
```
Comment (ytmimi) [inline Configurations.md:539]: Why did this need to move?

File: Configurations.md [hunk 2]
```diff
@@ -945,15 +945,15 @@ This option is currently ignored for stdin (`@generated` in stdin is ignored.)
...
+Format the bodies of macros.
 
-- **Default value**: `false`
+- **Default value**: `true`
 - **Possible values**: `true`, `false`
-- **Stable**: No (tracking issue: [#3354](https://github.com/rust-lang/rustfmt/issues/3354))
+- **Stable**: No (tracking issue: [#3355](https://github.com/rust-lang/rustfmt/issues/3355))
 
-#### `false` (default):
+#### `true` (default):
```
Comment (ytmimi) [inline Configurations.md:956]: Also assuming these changes were to put these options in alphabetical order?

File: Configurations.md [hunk 3]
```diff
@@ -1429,7 +1577,7 @@ fn main() {
 }
 ```
 
-See also: [`struct_lit_single_line`](#struct_lit_single_line), [`indent_style`](#indent_style).
+See also: [`struct_lit_single_line`](#struct_lit_single_line).
```
Comment (ytmimi) [inline Configurations.md:1580]: Why was `[indent_style](#indent_style)` removed?

Comment (ytmimi) [reply Configurations.md:1580]: Oh I see because this is the `indent_style` configuration section.

**Comment by ytmimi** [comment]
PR looks good to me.

The only thing I can recommend adding would be a test to ensure the configuration options stay in alphabetical order moving forward. We might consider adding the following test to `src/test/configuration_snippet.rs`

```rust
#[test]
fn check_config_options_in_alphabetical_order() {
    // Ensure the configs are listed in alphabetical order
    let lines = read_config_file_lines(CONFIGURATIONS_FILE_NAME);
    let configs: Vec<String> = lines
        .filter_map(|s| s.strip_prefix("## ").map(|s| s.to_owned()))
        .collect();

    let mut sorted_configs = configs.clone();
    sorted_configs.sort();

    assert!(
        configs == sorted_configs,
        "configs listed in {} are not in alphabetical order",
        CONFIGURATIONS_FILE_NAME
    );
}
```

Note: `read_config_file_lines` doesn't actually exist in the test suite, but would extract the logic for reading the configuration file, which is duplicated in both `check_unstable_option_tracking_issue_numbers` and `get_code_blocks`. I imagine something like this:

```rust
fn read_config_file_lines(name: &str) -> impl Iterator<Item = String> {
    BufReader::new(
        fs::File::open(Path::new(name)).unwrap_or_else(|_| panic!("couldn't read file {}", name)),
    )
    .lines()
    .map(Result::unwrap)
}
```

and then we could replace the duplicate code in the functions listed above with `read_config_file_lines`.

**Comment by ytmimi** [comment]
@TheTechn0mancer when you have a moment could you rebase and resolve the merge conflict.

**Comment by ytmimi** [comment]
From the command line
```
# add the rustfmt repo as a remote for your local repo
git remote add upstream https://github.com/rust-lang/rustfmt.git

# switch to your local patch-1 branch and rebase on the upstream master
git switch patch-1 
git pull --rebase upstream master

# fix the merge conflict
# You can follow steps 4-6 listed here:
# https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-using-the-command-line

# after fixing the merge conflicts add the changes you want to make
git add Configurations.md
git rebase --continue

# force push your changes back to your fork
git push origin patch-1 -f
```

**Comment by TheTechn0mancer** [comment]
I don't have a local repo, I just did the edits on GitHub's website interface.

**Pull request closed without merging**